### ISSUE TITLE
How could I create new mqtt clients in messageArrived?

### ISSUE THREAD

**Issue by forstef** [eclipse-paho/paho.mqtt.c#1109]
In my condition, I need to create new mqtt clients and connnect to server, accordding to revieved messages from messageArrived callback.
Could I put MQTTClient_create or MQTTClient_connect5 in messageArrived callback?
If not, why?

I have tried that, it look like MQTTClient_connect5 blocked.
Need I start a new thread to create another mqtt client and connect?

void CreateAnotherMQTTClient()
{
MQTTClient_create(...);
MQTTClient_connect5(...);
}
int messageArrived(void* context, char* topicName, int topicLen, MQTTClient_message* message)
{
std:: thread t(CreateAnotherMQTTClient())
t.detach();
//t.join(); //I tried this, but still blocked me.

MQTTClient_freeMessage(&message);
MQTTClient_free(topicName);
return 1;
}

ps: I uses #include "MQTTClient.h".

OR Just put the messages into some queue, and use another thread to process messages from queue?

**Comment by icraggs** [comment]
No.  You can't use other Paho C calls in a callback except for connect in connectionLost.  And especially not create.   You can put calls into callbacks in the MQTTAsync API, but even then *not* create.

**Comment by forstef** [comment]
@icraggs OK,Thanks.

**Comment by icraggs** [comment]
Answered.
